projects/gnosis-protocol/index.js

```diff
@@ -0,0 +1,97 @@
+/*==================================================
+  Modules
+==================================================*/
+
+const { api: { abi, eth, util }, util: utilBase } = require('../../sdk');
+
+/*==================================================
+  Settings
+==================================================*/
+
+const START_BLOCK = 9340147;
+const PROTOCOL_ADDRESS = '0x6F400810b62df8E13fded51bE75fF5393eaa841F';
+/* 
+* Tokens on GP to ignore
+[ 
+    0x7cac16770dd5f2a59859a395a492224f05a846b6: "oETH Put $200 29/05/2020",
+]
+*/
+const TOKENS_TO_IGNORE = new Set(['0x7cac16770dd5f2a59859a395a492224f05a846b6']);
+
+/*==================================================
+  Helper Functions
+==================================================*/
+
+const getTokenAddressFromLogData = data => '0x' + data.substring(26, 66);
+
+const getCallDataOfErc20Token = (tokenAddress, atThisBlock) => 
+  ({ 
+    target: tokenAddress, 
+    params: PROTOCOL_ADDRESS, 
+    abi: 'erc20:balanceOf', 
+    block: atThisBlock, 
+  });
+
+const getEthBalance = async block => eth.getBalance({ target: PROTOCOL_ADDRESS, block });
+
+/*==================================================
+  Main
+==================================================*/
+
+async function tvl(_, block) {
+  // Start ETH (read: wETH) balance check promise
+  const promisedEthBalance = getEthBalance(block);
```

Comment:
Unlike some other DeFi products, Gnosis Protocol doesn't support native ETH. We only support WETH which will be fetched as part of the token listing below. Therefore I think this value can safely supposed to be 0.

Reply:
internally this grabs WETH - I checked this in test when I manually had concatted WETH and found exactly 2x the ETH balance calculated

Reply:
deleted previous comment - you're right. re-checking